### PR TITLE
Updated toolbar download button style

### PR DESCRIPTION
**Pull request by simonhong**
Normal state -
Screenshot 2023-03-28 at 9 13 57 AM

In-progress -
Screenshot 2023-03-28 at 9 14 10 AM

Completed and user doesn't click the button yet.
Screenshot 2023-03-28 at 9 14 30 AM
fix https://github.com/brave/brave-browser/issues/29324

Resolves 

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: patches/chrome-browser-ui-views-download-bubble-download_toolbar_button_view.cc.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.cc b/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.cc
+index 7d3b2a36ed999db404460e315201f2ef37443f91..1976fe65c7b46e16d1244805c52f82e1a05d188d 100644
+--- a/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.cc
++++ b/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.cc
+@@ -59,7 +59,7 @@
+ 
+ namespace {
+ 
+-constexpr int kProgressRingRadius = 9;
++constexpr int kProgressRingRadius = 12;
```
Comment (simonhong) [inline patches/chrome-browser-ui-views-download-bubble-download_toolbar_button_view.cc.patch:10]: Made patch here. I think this is the most simplest way.
As this value is used in the middle of `PaintButtonContents()`,
we need to override that method and copy the most of that method's impls.

File: chromium_src/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_VIEWS_DOWNLOAD_BUBBLE_DOWNLOAD_TOOLBAR_BUTTON_VIEW_H_
+
+#define DownloadToolbarButtonView DownloadToolbarButtonViewChromium
+#define GetIconColor                                      \
+  UnUsed() {                                              \
+    return SK_ColorTRANSPARENT;                           \
+  }                                                       \
+                                                          \
+ protected:                                               \
+  DownloadDisplayController* display_controller() const { \
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.h:16]: Added to call this from the const method.

Comment (goodov) [reply chromium_src/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.h:16]: nit: returning mutable from `const` is not great, so another option here is to implement this as `DownloadDisplayController::IconInfo GetIconInfo() const { return controller_->GetIconInfo(); }`.

**Review by goodov — APPROVED** [review]

File: chromium_src/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.cc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#include "brave/browser/ui/color/brave_color_id.h"
+
+#include "chrome/browser/ui/views/download/bubble/download_toolbar_button_view.h"
+
+#define DownloadToolbarButtonView DownloadToolbarButtonViewChromium
+
+#include "src/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.cc"
+
+#undef DownloadToolbarButtonView
+
```
Comment (sangwoo108) [inline chromium_src/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.cc:15]: I'm not sure if it's worth, but can we remove the patch file by overriding gfx::RectFToSkRect() like this?

```suggestion
#include "ui/gfx/geometry/skia_conversions.h"

namespace gfx {
SkRect AdjustRingBounds(const gfx::RectF& ring_bounds);
}

#define RectFToSkRect(ring_bounds) AdjustRingBounds(ring_bounds)
#define DownloadToolbarButtonView DownloadToolbarButtonViewChromium

#include "src/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.cc"

#undef DownloadToolbarButtonView
#undef RectFToSkRect

namespace gfx {

SkRect AdjustRingBounds(const gfx::RectF& ring_bounds) {
  const int chromium_ring_radius = ui::TouchUiController::Get()->touch_ui()
                        ? kProgressRingRadiusTouchMode
                        : kProgressRingRadius;
  constexpr auto kBraveRingRadius = 12;
  const auto delta = kBraveRingRadius - chromium_ring_radius;
  gfx::RectF new_bounds(ring_bounds);
  new_bounds.Outset(gfx::OutsetsF(delta));
  return gfx::RectFToSkRect(new_bounds);
}

}  // namespace gfx
```

File: chromium_src/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.h (hunk 2 above)
Comment (sangwoo108) [reply chromium_src/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.h:16]: I think the return type should be `const DownloadDisplayController*` based on https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/const.md - The returned member could be changed so this getter doesn't seem to be logically const.

---

But changing return type like so would make us use const_cast<>, so I think we can remove this getter here and just use const_cast<> from the callsite.

Comment (sangwoo108) [reply chromium_src/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.h:16]: I just saw @goodov right after I wrote the comment above. +1 for @goodov 's suggestion.

Comment (simonhong) [reply chromium_src/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.h:16]: Thanks both! https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/const.md is nice doc :)

**Review by sangwoo108 — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.52.2 Chromium: 112.0.5615.39 (Official Build) nightly (64-bit)
-- | --
Revision | a0e7b9718a92bcd1cf33b7c95316caff3fc20714-refs/branch-heads/5615@{#753}
OS | Windows 11 Version 22H2 (Build 22621.1413)
```

Using the examples mentioned via https://github.com/brave/brave-core/pull/17769#issue-1641484433 & https://github.com/brave/brave-browser/issues/29324#issue-1641483155, verified the following:

**`Light Theme`**

defaultLight
downloadingLight
downloadProgressPanelLight
completedLight
completedIconLight

**`DarkTheme`**

defaultDark
image
downloadProgressPanelDark
completedDark
completedIconDark

**Pull request merged**